apps/main/src/dex/components/PageRouterSwap/index.tsx

```diff
@@ -98,30 +97,54 @@ const QuickSwap = ({
 
   const isReady = pageLoaded && !isLoadingApi && isPageVisible
   const haveSigner = !!signerAddress
+
   const userFromBalance = userBalancesMapper[fromAddress]
   const userToBalance = userBalancesMapper[toAddress]
+
   const fromUsdRate = usdRatesMapper[fromAddress]
   const toUsdRate = usdRatesMapper[toAddress]
-  const fromToken = tokensNameMapper[fromAddress] ?? ''
-  const toToken = tokensNameMapper[toAddress] ?? ''
 
   const tokensMapperNonSmallTvlStr = useMemo(
     () => getTokensMapperStr(tokensMapperNonSmallTvl),
     [tokensMapperNonSmallTvl],
   )
 
-  const selectFromTokensList = useMemo(
-    () => getTokensObjList(selectFromList ?? selectToList, tokensMapper),
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [selectFromList, selectToListStr, tokensMapperStr],
+  function getTokensObjList(tokensList: string[] | undefined, tokensMapper: TokensMapper | undefined) {
+    if (!tokensList || tokensList.length === 0 || !tokensMapper || Object.keys(tokensMapper).length === 0) return []
```

Comment:
```suggestion
    // import { isEmpty } from 'lodash'
    if (isEmpty(tokensList) || isEmpty(tokensMapper)) return []
```

Reply:
Had to add an exclamation mark to `return tokensList!.map()` as isEmpty works with `undefined`, but doesn't alter the type of `tokensList` from `string[] | undefined` to `string[]` even though we know for sure it's not undefined.